lib/usdUfe/ufe/UsdContextOps.cpp

```diff
@@ -0,0 +1,480 @@
+            varSets.GetNames(&varSetsNames);
+
+            if (itemPath.size() == 1u) {
+                // Variant sets list.
+                for (auto i = varSetsNames.crbegin(); i != varSetsNames.crend(); ++i) {
+                    items.emplace_back(*i, *i, Ufe::ContextItem::kHasChildren);
+                }
+            } else {
+                // Variants of a given variant set.  Second item in the path is
+                // the variant set name.
+                assert(itemPath.size() == 2u);
+
+                UsdVariantSet varSet = varSets.GetVariantSet(itemPath[1]);
+                auto          selected = varSet.GetVariantSelection();
+
+                const auto varNames = varSet.GetVariantNames();
+                for (const auto& vn : varNames) {
+                    const bool checked(vn == selected);
+                    items.emplace_back(
+                        vn,
+                        vn,
+                        Ufe::ContextItem::kNoChildren,
+                        Ufe::ContextItem::kCheckable,
+                        checked,
+                        Ufe::ContextItem::kExclusive);
+                }
+            } // Variants of a variant set
+        }     // Variant sets
+        else if (itemPath[0] == kUSDAddNewPrimItem) {
+            if (itemPath.size() == 1u) { // Root setup
+                items.emplace_back(
+                    kUSDDefPrimItem, kUSDDefPrimLabel, kUSDDefPrimImage); // typeless prim
+                items.emplace_back(kUSDScopePrimItem, kUSDScopePrimLabel, kUSDScopePrimImage);
+                items.emplace_back(kUSDXformPrimItem, kUSDXformPrimLabel, kUSDXformPrimImage);
+                items.emplace_back(Ufe::ContextItem::kSeparator);
+                items.emplace_back(kUSDCapsulePrimItem, kUSDCapsulePrimLabel, kUSDCapsulePrimImage);
+                items.emplace_back(kUSDConePrimItem, kUSDConePrimLabel, kUSDConePrimImage);
+                items.emplace_back(kUSDCubePrimItem, kUSDCubePrimLabel, kUSDCubePrimImage);
+                items.emplace_back(
+                    kUSDCylinderPrimItem, kUSDCylinderPrimLabel, kUSDCylinderPrimImage);
+                items.emplace_back(kUSDSpherePrimItem, kUSDSpherePrimLabel, kUSDSpherePrimImage);
+                items.emplace_back(Ufe::ContextItem::kSeparator);
+                items.emplace_back(
+                    kAllRegisteredTypesItem,
+                    kAllRegisteredTypesLabel,
+                    Ufe::ContextItem::kHasChildren);
+            } else if (itemPath.size() == 2u) { // Sub Menus
+                if (itemPath[1] == kAllRegisteredTypesItem) {
+                    // List the Registered schema plugins
+                    // Load this each time the menu is called in case plugins were loaded
+                    //      in between invocations.
+                    // However we cache it so the submenus don't need to re-query
+                    schemaTypeGroups = getConcretePrimTypes(true, getSchemaPluginNiceNames());
```

Comment:
Added the call the new helper here to send to other helper function to get nice names.